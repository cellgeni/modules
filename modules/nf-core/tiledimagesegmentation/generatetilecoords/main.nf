process TILEDIMAGESEGMENTATION_GENERATETILECOORDS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "quay.io/cellgeni/imagetileprocessor:0.1.11"

    input:
    tuple val(meta), path(image)

    output:
    tuple val(meta), path("${output_name}"), emit: tile_coords
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_tile_coords.csv"
    """
    tile-2d-image run \\
        --image ${image} \\
        --output_name "${output_name}" \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tile-2d-image: \$(tile-2d-image version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_tile_coords.csv"
    """
    touch "${output_name}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tile-2d-image: \$(tile-2d-image version)
    END_VERSIONS
    """
}
