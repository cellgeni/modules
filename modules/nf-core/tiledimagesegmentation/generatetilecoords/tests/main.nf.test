// nf-core modules test tiledimagesegmentation/generatetilecoords
nextflow_process {

    name "Test Process TILEDIMAGESEGMENTATION_GENERATETILECOORDS"
    script "../main.nf"
    config "./nextflow.config"
    process "TILEDIMAGESEGMENTATION_GENERATETILECOORDS"

    tag "modules"
    tag "modules_nfcore"
    tag "tiledimagesegmentation"
    tag "tiledimagesegmentation/generatetilecoords"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test',], // meta map
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
