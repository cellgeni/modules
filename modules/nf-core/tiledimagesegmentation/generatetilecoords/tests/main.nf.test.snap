{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,20ba3b70d4bdccb62670a9d99bb2ea35"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,20ba3b70d4bdccb62670a9d99bb2ea35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T14:00:07.23102576"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,20ba3b70d4bdccb62670a9d99bb2ea35"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,20ba3b70d4bdccb62670a9d99bb2ea35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T13:59:58.301030522"
    }
}